components/public-api/gitpod/v1/workspace.proto

```diff
@@ -38,6 +132,233 @@ message WatchWorkspaceStatusResponse {
   WorkspaceStatus status = 2;
 }
 
+message ListWorkspacesRequest {
+  enum OrderByField {
+    ORDER_BY_FIELD_UNSPECIFIED = 0;
+    // ORDER_BY_FIELD_CREATION_TIME is the default sorting field, it sorts
+    // workspaces by (instances) creation time
+    ORDER_BY_FIELD_CREATION_TIME = 1;
+  }
+
+  // pagination contains the pagination options for listing workspaces
+  PaginationRequest pagination = 1;
+
+  // order_by is the field to sort workspaces by, allowed values are OrderBy
+  OrderBy order_by = 2;
+
+  // scope of the workspaces
+  WorkspaceScope scope = 3;
+
+  // organization_id is the ID of the organization that contains the workspaces
+  string organization_id = 4;
+
+  // pinned indicates whether to list only pinned workspaces
+  bool pinned = 5;
+
+  // search_term is a search term to filter workspaces by name
+  string search_term = 6;
+
+  string workspace_id = 7;
+
+  string owner_id = 8;
+}
+
+message ListWorkspacesResponse {
+  // workspaces are the workspaces that matched the query
+  repeated Workspace workspaces = 1;
+
+  // pagination contains the pagination options for listing workspaces
+  PaginationResponse pagination = 2;
+}
+
+message CreateAndStartWorkspaceRequest {
+  message Source {
+    string context_url = 1;
+    string repository_id = 2;
+  }
+
+  string organization_id = 1;
+
+  Source source = 2;
+  StartWorkspaceSpec spec = 3;
+
+  bool ignore_running_workspace_on_same_commit = 4;
+  bool ignore_running_prebuild = 5;
+  bool allow_using_previous_prebuilds = 6;
+  bool force_default_config = 7;
+}
+
+message CreateAndStartWorkspaceResponse { string workspace_id = 1; }
+
+message StartWorkspaceRequest { string workspace_id = 1; }
+
+message StartWorkspaceResponse {}
+
+message StopWorkspaceRequest {
+  string workspace_id = 1;
+
+  // scope of the workspace to stop
+  WorkspaceScope scope = 2;
+}
+
+message StopWorkspaceResponse {}
+
+message DeleteWorkspaceRequest { string workspace_id = 1; }
+
+message UpdateWorkspaceRequest {
+
+  string workspace_id = 1;
+
+  WorkspaceSpec spec = 2;
```

Comment:
It should not be like that, we should list here optional updatable files, please read guidelines carefully.